src/lucky_record/errors.cr

```diff
@@ -0,0 +1,20 @@
+# = Lucky Record Errors
+#
+# Generic Lucky Record exception class.
+class LuckyRecordError < Exception
+end
+
+# Raised when Lucky Record cannot find a record by given id
+class RecordNotFound < LuckyRecordError
+  getter model : (String|Nil)
+  getter primary_key : (String|Nil)
+  getter id : (String|Int32|Nil)
```

Comment:
I think these should not be nilable. That means you'd need to pass them in when you instantiate the Error. Also I think we can remove the `primary_key` field for now since it is always `id`